binary_gentoo/internal/cli/tree_diff.py

```diff
@@ -22,11 +82,23 @@ def iterate_new_and_changed_ebuilds(config):
         for ebuild_file in ebuild_files:
             old_portdir_ebuild_file = os.path.join(config.old_portdir, category_plus_package,
                                                    ebuild_file)
+            new_portdir_ebuild_file = os.path.join(root, ebuild_file)
+
+            # don't output 9999 ebuilds
+            if ebuild_file.endswith('-9999.ebuild'):
+                continue
+
+            # don't output if files are identical
             if os.path.exists(old_portdir_ebuild_file):
-                new_portdir_ebuild_file = os.path.join(root, ebuild_file)
                 if filecmp.cmp(old_portdir_ebuild_file, new_portdir_ebuild_file):
                     continue
 
+            # include only specific keywords sets
+            if not _keywords_included_in_ebuild(
+                    config.keywords, new_portdir_ebuild_file,
+                    old_portdir_ebuild_file if not config.report_changes else None):
```

Comment:
The `old_portdir_ebuild_file if not config.report_changes else None` is a nice trick on one hand, but does not easily explain itself.

Reply:
Correct - removed because code has been reorganized